lib/cisco_node_utils/bgp.rb

```diff
@@ -338,23 +339,35 @@ def default_enforce_first_as
 
     # Confederation Peers (Getter/Setter/Default)
     def confederation_peers
-      config_get('bgp', 'confederation_peers', @get_args)
-    end
-
-    def confederation_peers_set(peers)
-      # The confederation peers command is additive so we first need to
-      # remove any existing peers.
-      unless confederation_peers.empty?
-        @set_args[:state] = 'no'
-        @set_args[:peer_list] = confederation_peers
-        config_set('bgp', 'confederation_peers', @set_args)
-      end
-      unless peers == default_confederation_peers
-        @set_args[:state] = ''
-        @set_args[:peer_list] = peers
-        config_set('bgp', 'confederation_peers', @set_args)
+      # NX: confederation peers is retrieved as a string '1 2 3'
+      # XR: retrieved as an array.
+      # So make it an array in the NX case.
+      # Sort the end results to make it consistent across both platforms.
+      # And if peers is an empty list - make it an empty array.
+      peers = config_get('bgp', 'confederation_peers', @get_args)
+      peers = peers.split(' ') if peers.is_a?(String)
+      peers.empty? ? [] : peers.sort
```

Comment:
Is the empty check needed, or does `peers.sort` work just fine even if the array is empty?